bentoml/adapters/annotated_image_input.py

```diff
@@ -0,0 +1,384 @@
+                        "'application/json' file.  Alternatively, it can accept two "
+                        "'application/octet-stream' files, one with a valid image "
+                        "extension and one with a '.json' extension respectively"
+                    )
+            else:
+                raise BadInput(
+                    "BentoML#AnnotatedImageInput unexpected HTTP request format"
+                )
+
+        if not image_file:
+            raise BadInput("BentoML#AnnotatedImageInput requires an image file")
+
+        image_file_name = secure_filename(image_file.filename)
+        verify_image_format_or_raise(image_file_name, self.accept_image_formats)
+        input_stream = image_file.stream
+        input_image = imageio.imread(input_stream, pilmode=self.pilmode)
+
+        if json_file:
+            input_json = read_json_file(json_file)
+            return {
+                self.image_input_name: input_image,
+                self.annotation_input_name: input_json,
+            }
+
+        return {self.image_input_name: input_image}
+
+    def handle_batch_request(
+        self, requests: Iterable[SimpleRequest], func: callable
+    ) -> Iterable[SimpleResponse]:
+        """
+        Batch version of handle_request
+        """
+        input_datas = []
+        slices = []
+
+        for i, req in enumerate(requests):
+            if not req.data:
+                slices.append(None)
+                input_datas.append(None)
+                continue
+            request = Request.from_values(
+                input_stream=BytesIO(req.data),
+                content_length=len(req.data),
+                headers=req.headers,
+            )
+            try:
+                input_data = self._load_image_and_json_data(request)
+            except BadInput:
+                slices.append(None)
+                input_datas.append(None)
+                continue
+
+            input_datas.append(input_data)
+            slices.append(i)
+
+        results = [func(**d) if d else {} for d in input_datas]
+
+        return self.output_adapter.to_batch_response(
+            result_conc=results,
+            slices=slices,
+            fallbacks=[None] * len(slices),
+            requests=requests,
+        )
+
+    def handle_request(self, request, func):
+        """Handle http request that has one image file. It will convert image into
+        a ndarray for the function to consume
+
+        Args:
+            request: incoming request object.
+            func: function that will take ndarray as its arg.
+            options: configuration for handling request object.
+        Return:
+            response object
+        """
+        input_data = self._load_image_and_json_data(request)
+        result = func(**input_data)
+        return self.output_adapter.to_response(result, request)
+
+    def handle_cli(self, args, func):
```

Comment:
This API `handle_cli` may change in the next version, adding support for batch input and other common logic for each adapter.

Reply:
Got it, thanks for the heads up.